File: api/src/main/java/com/beco/api/model/dto/ConstructionSiteDto.java
package com.beco.api.model.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.Instant;
import java.util.Date;
import java.util.UUID;

@Data
public class ConstructionSiteDto {

    private UUID constructionSiteId;

    private CustomerDto constructionSiteCustomer;

    private GetAddressDto constructionSiteAddress;

    private SharedDetailsDto sharedDetails;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date dateStart;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private Date dateEnd;

    private Instant createdDate;

    private Instant lastModifiedDate;
}

File: spa/src/app/core/model/dto/construction-site.dto.ts
import { CustomerDto } from './customer.dto';
import { SharedDetailsDto } from './shared-details.dto';
import {AddressDto} from './address.dto';

export interface ConstructionSiteDto {
  constructionSiteId: string;
  constructionSiteCustomer: CustomerDto;
  constructionSiteAddress: AddressDto;
  sharedDetails: SharedDetailsDto;
  dateStart: string; // yyyy-MM-dd
  dateEnd: string;   // yyyy-MM-dd
  createdDate: string;
  lastModifiedDate: string;
}
